mkosi/distributions/debian.py

```diff
@@ -28,67 +26,101 @@ def kernel_image(name: str, architecture: str) -> Path:
     def initrd_path(kver: str) -> Path:
         return Path("boot") / f"initrd.img-{kver}"
 
-    @classmethod
-    def install(cls, state: MkosiState) -> None:
-        repos = {"main", *state.config.repositories}
-
-        cmdline: list[PathString] = [
-            "debootstrap",
-            "--variant=minbase",
-            "--merged-usr",
-            f"--cache-dir={state.cache.absolute()}",
-            f"--components={','.join(repos)}",
-        ]
-
-        debarch = DEBIAN_ARCHITECTURES[state.config.architecture]
-        cmdline += [f"--arch={debarch}"]
-
-        # Let's use --no-check-valid-until only if debootstrap knows it
-        if debootstrap_knows_arg("--no-check-valid-until"):
-            cmdline += ["--no-check-valid-until"]
-
-        if not state.config.repository_key_check:
-            cmdline += ["--no-check-gpg"]
-
-        mirror = state.config.local_mirror or state.config.mirror
-        assert mirror is not None
-        cmdline += [state.config.release, state.root, mirror]
-
-        # Pretend we're lxc so debootstrap skips its mknod check.
-        run_with_apivfs(state, cmdline, env=dict(container="lxc", DPKG_FORCE="unsafe-io"))
-
-        # systemd-boot won't boot unified kernel images generated without a BUILD_ID or VERSION_ID in
-        # /etc/os-release. Build one with the mtime of os-release if we don't find them.
-        with state.root.joinpath("etc/os-release").open("r+") as f:
-            os_release = f.read()
-            if "VERSION_ID" not in os_release and "BUILD_ID" not in os_release:
-                f.write(f"BUILD_ID=mkosi-{state.config.release}\n")
-
-        if not state.config.local_mirror:
-            cls._add_apt_auxiliary_repos(state, repos)
+    @staticmethod
+    def repositories(state: MkosiState, local: bool = True) -> list[str]:
+        repos = ' '.join(("main", *state.config.repositories))
+
+        if state.config.local_mirror and local:
+            return [f"deb [trusted=yes] {state.config.local_mirror} {state.config.release} {repos}"]
+
+        main = f"deb {state.config.mirror} {state.config.release} {repos}"
+
+        if state.config.release in ("unstable", "sid"):
+            return [main]
+
+        updates = f"deb {state.config.mirror} {state.config.release}-updates {repos}"
+
+        # Security updates repos are never mirrored
+        if state.config.release in ("stretch", "buster"):
+            security = f"deb http://security.debian.org/debian-security/ {state.config.release}/updates {repos}"
         else:
-            # Add a single local offline repository, and then remove it after apt has ran
-            state.root.joinpath("etc/apt/sources.list.d/mirror.list").write_text(f"deb [trusted=yes] {state.config.local_mirror} {state.config.release} main\n")
+            security = f"deb https://security.debian.org/debian-security {state.config.release}-security {repos}"
```

Comment:
```
root@image:~# apt update
Err:1 https://security.debian.org/debian-security stable-security InRelease
  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 2a04:4e42:600::644 443]
Get:2 http://deb.debian.org/debian stable InRelease [116 kB]
Get:3 http://deb.debian.org/debian stable-updates InRelease [44.1 kB]
Get:4 http://deb.debian.org/debian stable/main amd64 Packages [8183 kB]
Get:5 http://deb.debian.org/debian stable/main Translation-en [6240 kB]
Get:6 http://deb.debian.org/debian stable-updates/main amd64 Packages [14.6 kB]
Get:7 http://deb.debian.org/debian stable-updates/main Translation-en [7929 B]
Fetched 14.6 MB in 2s (7310 kB/s)                               
Reading package lists... Done
Building dependency tree... Done
All packages are up to date.
W: https://security.debian.org/debian-security/dists/stable-security/InRelease: No system certificates available. Try installing ca-certificates.
W: Failed to fetch https://security.debian.org/debian-security/dists/stable-security/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 2a04:4e42:600::644 443]
W: Some index files failed to download. They have been ignored, or old ones used instead.
```

Given https is used here, it also needs to add ca-certificates to the package list, otherwise it will not work

Reply:
Shouldn't we use the cert store from the host? I wonder if we're missing some apt config

Reply:
I just switched to http since we use http for the default debian and ubuntu mirrors as well.